tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/caches/IntRangeCache.scala

```diff
@@ -0,0 +1,45 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.aux.caches
+
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.bmcmt.types.CellT
+import at.forsyte.apalache.tla.bmcmt.{ArenaCell, FixedElemPtr, PureArena}
+import at.forsyte.apalache.tla.lir.{IntT1, SetT1}
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * Cache tuple domains as well as ranges a..b.
```

Comment:
Not sure, but isn't one just a special case of the other?

```suggestion
 * Cache ranges a..b and, as a special case, tuple domains.
```